src/app/language-settings/language-settings.ts

```diff
@@ -217,4 +221,12 @@ export class LanguageSettingsPage implements OnInit {
       this.commonUtilService.showToast(dom.body.textContent, false, 'redErrorToast');
     }
   }
+
+  handleHeaderEvents($event) {
+    if ($event.name === 'back') {
+      this.telemetryGeneratorService.generateBackClickedTelemetry(
+          PageId.SETTINGS_LANGUAGE, Environment.HOME, true);
```

Comment:
env and page id should be dynamic based on  this.isFromSettings.
 this.isFromSettings ? Environment.SETTINGS : Environment.ONBOARDING,
      this.isFromSettings ? PageId.SETTINGS : PageId.ONBOARDING_LANGUAGE_SETTING